### ISSUE TITLE
Bug in EmptyLineBetweenDefs

### ISSUE THREAD

**Issue by bquorning** [bbatsov/rubocop#2177]
Conditional method definitions are reported as offenses in the latest master (9765aee09f8f7423ec21fbb4ddcb6f34857ca39e):
```rb
# encoding: utf-8
if RUBY_PLATFORM == '2.2.2'
  def foo
    'A'
  end
else
  def foo
    'B'
  end
end
```

```
test.rb:7:3: C: Use empty lines between defs.
  def foo
  ^^^

1 file inspected, 1 offense detected
```

Git bisect tells me that this behavior was introduced by #2158.

By the way, autocorrecting the code adds one just blank line, and re-running produces no RuboCop offenses:
```rb
# encoding: utf-8
if RUBY_PLATFORM == '2.2.2'
  def foo
    'A'
  end

else
  def foo
    'B'
  end
end
```
That doesn’t look right.

cc @unmanbearpig

**Comment by unmanbearpig** [comment]
Yeah, it looks like a problem I might've introduced, I'll look into it a bit later today.

**Comment by bquorning** [comment]
Fixed by #2181.
